### PR TITLE
[Docs] Update Development Environment Setup Guide for MonoRepo

### PR DESCRIPTION
**Pull request by Future-Outlier**
## Doc link
https://flyte--4257.org.readthedocs.build/en/4257/community/contribute.html

## Tracking issue
https://github.com/flyteorg/flyte/issues/4188

## Describe your changes
- Update the "hello world" example in flytesnacks
- Update the setup guide to reflect the Monorepo structure.
- Update the flytekit install plugins guideline, since flytekitplugins doesn't support lazy import.

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] All commits are signed-off.

## Screenshots
image
image

## Note to reviewers
@Yicheng-Lu-llll, @samhita-alla, @pingsutw please take a look, thanks a lot!

### CODE REVIEW

File: rsts/community/contribute.rst [hunk 1]
```diff
@@ -542,8 +540,8 @@ If not, we can start backends with a single command.
    # Step3: Run a hello world sample to test locally
    git clone https://github.com/flyteorg/flytesnacks
    cd flytesnacks
-   python3 examples/basics/basics/hello_world.py my_wf
-   # Running my_wf() hello world
+   python3 examples/basics/basics/hello_world.py hello_world_wf
```
Comment (pingsutw) [inline rsts/community/contribute.rst:543]:
```suggestion
   pyflyte run https://github.com/flyteorg/flytesnacks/blob/master/examples/basics/basics/hello_world.py hello_world_wf
```

File: rsts/community/contribute.rst [hunk 2]
```diff
@@ -586,7 +584,7 @@ the Flyte cluster, and finally submit the workflow.
    # Step4: Submit a hello world workflow to the Flyte cluster
    git clone https://github.com/flyteorg/flytesnacks
    cd flytesnacks
-   pyflyte run --image ${FLYTE_INTERNAL_IMAGE} --remote examples/basics/basics/hello_world.py my_wf
+   pyflyte run --image ${FLYTE_INTERNAL_IMAGE} --remote examples/basics/basics/hello_world.py hello_world_wf
```
Comment (pingsutw) [inline rsts/community/contribute.rst:587]:
```suggestion
   pyflyte run --image ${FLYTE_INTERNAL_IMAGE} --remote https://github.com/flyteorg/flytesnacks/blob/master/examples/basics/basics/hello_world.py hello_world_wf
```

File: rsts/community/contribute.rst [hunk 3]
```diff
@@ -531,9 +514,24 @@ If not, we can start backends with a single command.
    make setup
    pip install -e .
    pip install gsutil awscli
```
Comment (pingsutw) [inline rsts/community/contribute.rst:516]: could you remove this? we don't need it actually

File: rsts/community/contribute.rst (hunk 2 above)
Comment (pingsutw) [reply rsts/community/contribute.rst:587]: and remove the git clone

**Comment by samhita-alla** [comment]
@pingsutw, can this PR be merged now?

**Review by pingsutw — APPROVED** [review]

File: rsts/community/contribute.rst [hunk 4]
```diff
@@ -415,48 +415,31 @@ that integrates all Flyte components into a single binary.
...
-   # If you have modified any of these components/libraries, use `go mod edit -replace` in the {flyteadmin} repo to replace original components/libraries with your customized ones.
-   # For instance, if you have also modified `flytepropeller`, run the following commands:
-   go mod edit -replace github.com/flyteorg/flytepropeller=/home/ubuntu/flytepropeller #replace with your own local path to flytepropeller
+   # Step2: Go to the {flyteadmin} repository, modify the source code accordingly.
+   cd flyte/flyteadmin
 
-   # Step 3.2: Generate code, fix lint errors and run unit tests for {flyteadmin}.
-   # Note, flyteidl does not have unit tests, so you can skip the `make test_unit` command.
-   # Note, flytestdlib only have `make generate` command.
-   make generate
```
Comment (Yicheng-Lu-llll) [inline rsts/community/contribute.rst:452]: @Future-Outlier I think we still need to execute the following commands:
```shell
make generate
make lint
make test_unit
```
For instance, to generate the protobuf code, I guess we still need to run `make generate` under flyteidl folder? Correct me if I am wrong.

**Pull request merged**